README.md

```diff
@@ -1,27 +1,508 @@
+`BlogPost` component that we defined using an arrow function):
+
+```js
+class Blog extends React.Component {
+    componentDidMount () {
+        fetch('http://localhost:4000/api/posts')
+            .then(response => response.json())
+            .then(posts => {
+                console.log('Yey! I got some posts!', posts)
+            })
+    }
+    render () {
+        return (
+            <div>
+                <BlogPost title={'Hello World!'} body={'Hello, this is my blog.'} />
+                <BlogPost title={'Good bye!'} body={'I\'m busy. I\'m shutting this blog down.'} />
+            </div>
+        )
+    }
+}
+```
+
+This means that whenever we `mount` this component by rendering it as a child of
+some other part of our application, e.g. using `<Blog />`, React executes the
+`componentDidMount` method that requests data from our API.
+
+Once we receive those blog posts, we `console.log` them. While
+`console.log`ging them is certainly fun, our users won't be too happy about
+having to open the Chrome DevTools in order to read our blog. So instead of just
+printing them to the console, the `Blog` component should somehow "keep track
+of them".
+
+The most intuitive solution would be to "set" them as a property on the
+component. React provides a method for that called... `setState`! `setState`
+allows components to keep track of their **internal** state.
+
+So let's change our component to update the state of the blog component once we
+receive the requested blog posts from our API:
+
+```js
+    componentDidMount () {
+        fetch('http://localhost:4000/api/posts')
+            .then(response => response.json())
+            .then(posts => {
+                console.log('Yey! I got some posts!', posts)
+
+                // Let's update the state here:
+                this.setState({ posts: posts })
+
+                // or shorter: this.setState({ posts })
+            })
+    }
+```
+
+Now that we updated the component's state using `setState`, we want to render
+our posts using the `BlogPost` component, so we need to update our component's
+render method:
+
+```js
+    render () {
+        return (
+            <div>
+                {
+                    this.state.posts.map((post) =>
+                        <BlogPost title={post.title} body={post.body} />
+                    )
+                }
+            </div>
+        )
+    }
+```
+
+Keep in mind that we didn't change the `BlogPost` component *at all* — ideally
+you should try to keep the majority of your components "stateless".
+
+At this point, you might wonder why we had to use `setState`, if all `setState`
+does is... well... setting `this.state`. There is a very simple reason for that:
+
+React needs to "know" that a state change happened, since whenever a component's
+"data" (state or props) changes, React has to re-render the component (runs its
+render function to figure out "what changed").
+
+### State vs Props
```

Comment:
Students should have already learned the difference at this point, but I really like the idea of giving them a quick refresher (one or two paragraphs should do it). 👍